src/components/encoding-pane/index.tsx

```diff
@@ -43,15 +46,20 @@ class EncodingPanelBase extends React.PureComponent<EncodingPanelProps, {}> {
 
   public render() {
     const {specPreview, spec} = this.props;
+    const {manualSpecificationOnly} = this.props.config;
     const {anyEncodings} = specPreview || spec;
 
     const positionShelves = ['x', 'y'].map(this.encodingShelf, this);
     const facetShelves = ['row', 'column'].map(this.encodingShelf, this);
     const nonPositionShelves = ['size', 'color', 'shape', 'detail', 'text'].map(this.encodingShelf, this);
-    const wildcardShelves = [
-      ...anyEncodings.map((_, i) => i),
-      -1 // map the empty placeholder to -1
-    ].map(this.wildcardShelf, this);
+    const wildcardShelves = !manualSpecificationOnly ? (
+      <div>
```

Comment:
Why adding a new div instead of moving the whole shelf-group up here? 

I would do that (to avoid adding another level of `<div>`) and rename `wildcardShelves` to `wildcardShelvesGroup`